### DISCUSSION TITLE
Cut Keybinding Does not Work

### DISCUSSION THREAD

**Discussion by blueray453** [fish-shell/fish-shell#9832] [Q&A] [answered]
The following keybinding works.

    bind \cx fish_clipboard_copy

But the following does not work.

    bind \cx fish_clipboard_copy kill-selection

What might be the issue?

**Comment by faho** [comment, marked as answer]
From https://fishshell.com/docs/current/cmds/bind.html:

> Note

Special input functions cannot be combined with ordinary shell script commands. The commands must be entirely a sequence of special input functions (from bind -f) or all shell script commands (i.e., valid fish script). To run special input functions from regular fish script, use commandline -f (see also [commandline](https://fishshell.com/docs/current/cmds/commandline.html)). If a script produces output, it should finish by calling commandline -f repaint to tell fish that a repaint is in order.

`fish_clipboard_copy` is a normal script function.

**Comment by blueray453** [comment]
The following works:

```
function ctrl_x_cut
    set -l cmdline_selection
    set cmdline_selection (commandline --current-selection | string collect)
    # set cmdline (commandline --current-buffer | string collect)

    if test -n "$cmdline_selection"
        printf '%s' $cmdline_selection | xclip -selection clipboard
        commandline -f kill-selection
        commandline -f end-selection
    end
end
```
